File: sub-modules/backup/LinkManager/model/Node.go

package model

const (
	MAX_INSTANCE_NODE = 128
	MASTER_NODE_MAKEUP = 16
) 

type Node struct {
	NodeID uint32
	FreeInstance int
	IsMasterNode bool
	L3AddrV4 uint32
	L3AddrV6 uint64
	L2Addr uint64 // 低六字节储存MAC地址
	NsInstanceMap map[string]string
	NsLinkMap map[string]string
}

File: sub-modules/PositionProvider/opensn/model/node.py
import json
class Node:
    
    def __init__(self) -> None:
        self.node_index: int = 0
        self.free_instance: int = 0
        self.is_master_node: bool = False
        self.l_3_addr_v_4: bytes = b''
        self.l_3_addr_v_6: bytes = b''
        self.l_2_addr: bytes = b''
        self.ns_instance_map: dict[str,str] = {}
        self.ns_link_map: dict[str,str] = {}
        self.node_link_device_info: dict[str,int] = {}

def node_from_json(seq: str) -> None:
    node = Node()
    if seq is None :
        return node
    node.__dict__ = json.loads(seq)
    return node